tests/Microsoft.Bot.Builder.TestBot.WebApi/Dialogs/DateResolverDialog.cs

```diff
@@ -16,16 +16,37 @@ public DateResolverDialog(string id = null)
             : base(id ?? nameof(DateResolverDialog))
         {
             AddDialog(new DateTimePrompt(nameof(DateTimePrompt), DateTimePromptValidator));
-            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
+            var steps = new WaterfallStep[]
             {
                 InitialStepAsync,
                 FinalStepAsync,
-            }));
+            };
+            AddDialog(new WaterfallDialog(nameof(WaterfallDialog), steps));
 
             // The initial child Dialog to run.
             InitialDialogId = nameof(WaterfallDialog);
         }
 
+        private static Task<bool> DateTimePromptValidator(PromptValidatorContext<IList<DateTimeResolution>> promptContext, CancellationToken cancellationToken)
```

Comment:
Why move this?
The original order was intentional. we can move it if you like but exactly why would we do that?
(and again consistency with the original sample)